### PR TITLE
4.1 TinyMCE source code  plugin clean up

### PR DESCRIPTION
**Pull request by dgrammatiko**
Pull Request for Issue # .

### Summary of Changes
- Tweaked the modal window size
- Moved files to ES6 so they can be linted
- Removed the `document.write`

### Testing Instructions

Check that the source code is still working as expected

### Documentation Changes Required
NO

@Fedik mainly this PR is an alternative to your commit for the removal of the `document.write` which I reverted as it wasn't working. Can you review this one?

ping @brianteeman @woluweb @bembelimen

### CODE REVIEW

**Comment by Fedik** [comment]
Looks good to me :)

**Comment by brianteeman** [comment]
I have tested this item :white_check_mark: successfully on 746488c0e451664aa8e0a6592d7dfd5fb3c9ee8f<hr /><sub>This comment was created with the <a href="https://github.com/joomla/jissues">J!Tracker Application</a> at <a href="https://issues.joomla.org/tracker/joomla-cms/36112">issues.joomla.org/tracker/joomla-cms/36112</a>.</sub>

**Comment by brianteeman** [comment]
Depending on the size of the screen this is miscaclulating the height of the codemirror window and as a result you cannot get to the top or the bottom 

tiny

**Comment by brianteeman** [comment]
Before this pr the window could be closed with the escape key. Now it has to be done from the button

**Comment by dgrammatiko** [comment]
> Before this pr the window could be closed with the escape key. Now it has to be done from the button

Which browser? Works fine in Chrome here

**Comment by brianteeman** [comment]
I found out why/when the escape key didnt work. If you click outside of the window then the escape doesnt work. Not sure if it should or shouldnt

**Comment by dgrammatiko** [comment]
> I found out why/when the escape key didnt work. If you click outside of the window then the escape doesnt work. Not sure if it should or shouldnt

Ah, that's expected (code wise) as the listener is added in the iframe document and when you click outside of the modal you change scope to the parent document (which doesn't have any code to handle the modal). Let me check something...

**Comment by dgrammatiko** [comment]
@brianteeman I restored the original sizing for the time. We could tackle this in another PR

**Comment by khu5h1** [comment]
I have tested this item :white_check_mark: successfully on 60bb10955772f23c4693a9c2536853cfe94cd0e4<hr /><sub>This comment was created with the <a href="https://github.com/joomla/jissues">J!Tracker Application</a> at <a href="https://issues.joomla.org/tracker/joomla-cms/36112">issues.joomla.org/tracker/joomla-cms/36112</a>.</sub>

File: build/media_source/plg_editors_tinymce/js/plugins/highlighter/source.es6.js [hunk 1]
```diff
@@ -195,85 +226,50 @@ const start = () => {
   CMsettings.config.value = html;
 
   // Instantiante CodeMirror:
```
Comment (PhilETaylor) [inline build/media_source/plg_editors_tinymce/js/plugins/highlighter/source.es6.js]:
```suggestion
  // Instantiate CodeMirror:
```

Comment (dgrammatiko) [reply build/media_source/plg_editors_tinymce/js/plugins/highlighter/source.es6.js]: Thanks

**Comment by laoneo** [comment]
I have tested this item :white_check_mark: successfully on 1153eea6c7df3ab0e09a2775c8fe26e11dee867b<hr /><sub>This comment was created with the <a href="https://github.com/joomla/jissues">J!Tracker Application</a> at <a href="https://issues.joomla.org/tracker/joomla-cms/36112">issues.joomla.org/tracker/joomla-cms/36112</a>.</sub>

**Comment by jwaisner** [comment]
I have tested this item :white_check_mark: successfully on 1153eea6c7df3ab0e09a2775c8fe26e11dee867b<hr /><sub>This comment was created with the <a href="https://github.com/joomla/jissues">J!Tracker Application</a> at <a href="https://issues.joomla.org/tracker/joomla-cms/36112">issues.joomla.org/tracker/joomla-cms/36112</a>.</sub>

**Comment by jwaisner** [comment]
RTC<hr /><sub>This comment was created with the <a href="https://github.com/joomla/jissues">J!Tracker Application</a> at <a href="https://issues.joomla.org/tracker/joomla-cms/36112">issues.joomla.org/tracker/joomla-cms/36112</a>.</sub>

**Comment by bembelimen** [comment]
Thx

**Pull request merged**